### PR TITLE
rename param_map to data in handlers

### PR DESCRIPTION
**Pull request by fehiepsi**
@neerajprad There is still `param_map` left in [ELBO.loss](http://num.pyro.ai/en/stable/svi.html#numpyro.infer.elbo.ELBO.loss). Should we rename it to `params` to make it consistent with `AutoGuide` or utilities in `infer/util.py`? I just notice when trying to grep `param_map` and have less opinion about it. WDYT?

### CODE REVIEW

File: numpyro/infer/util.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def potential_energy(model, model_args, model_kwargs, params):
 
 def _init_to_unconstrained_value(site=None, values={}):
     if site is None:
-        return partial(init_to_value, values=values)
+        return partial(_init_to_unconstrained_value, values=values)
```
Comment (fehiepsi) [inline numpyro/infer/util.py:154]: @neerajprad I think this is my original intention. My last fix is a bit misleading (though it still solves the issue)... :(

File: numpyro/distributions/discrete.py [hunk 2]
```diff
@@ -550,7 +550,7 @@ def __init__(self, rate, validate_args=None):
         super(Poisson, self).__init__(jnp.shape(rate), validate_args=validate_args)
 
     def sample(self, key, sample_shape=()):
-        return random.poisson(key, device_put(self.rate), shape=sample_shape + self.batch_shape)
+        return random.poisson(key, self.rate, shape=sample_shape + self.batch_shape)
```
Comment (fehiepsi) [inline numpyro/distributions/discrete.py:553]: we no longer need this in the last JAX release

File: numpyro/infer/util.py (hunk 1 above)
Comment (neerajprad) [reply numpyro/infer/util.py:154]: Thanks, now I understand what's happening. :)

File: numpyro/handlers.py [hunk 3]
```diff
@@ -241,12 +241,12 @@ def process_message(self, msg):
...
 class condition(Messenger):
     """
-    Conditions unobserved sample sites to values from `param_map` or `condition_fn`.
+    Conditions unobserved sample sites to values from `data` or `condition_fn`.
     Similar to :class:`~numpyro.handlers.substitute` except that it only affects
     `sample` sites and changes the `is_observed` property to `True`.
 
     :param fn: Python callable with NumPyro primitives.
-    :param dict param_map: dictionary of `numpy.ndarray` values keyed by
+    :param dict data: dictionary of `numpy.ndarray` values keyed by
```
Comment (neerajprad) [inline numpyro/handlers.py:249]: What do you think about raising a `FutureWarning` with `param_map`? We can remove all of these deprecated arguments in the next release.

Comment (fehiepsi) [reply numpyro/handlers.py:249]: Sure, I'll add a FutureWarning for this.
